src/main/kotlin/com/openlattice/data/storage/DataDeletionService.kt

```diff
@@ -248,32 +318,14 @@ class DataDeletionService(
 
     }
 
-    /** Association deletion and auth **/
-
-    // TODO do we need this??
-    private fun authorizeAndDeleteAssociationsForNeighborhood(
-            entitySetId: UUID,
-            entityKeyIds: Set<UUID>,
-            srcEntitySetFilter: Set<UUID>,
-            dstEntitySetFilter: Set<UUID>,
-            deleteType: DeleteType,
-            principals: Set<Principal>
-    ): Int {
-
-        // if no filters, just treat at a normal delete
-        if (srcEntitySetFilter.isEmpty() && dstEntitySetFilter.isEmpty()) {
-            return authorizeAndDeleteAssociations(entitySetId, Optional.of(entityKeyIds), deleteType, principals)
-        }
-
-
-
-
-
-        return 0
-    }
-
+    /* Association deletion and auth */
 
-    private fun authorizeAndDeleteAssociations(
+    /**
+     * Checks authorizations on connected association entities and clears/deletes them along with edges connected to
+     * them.
+     * If entity set is an association entity set, it only deletes its edges.
+     */
+    private fun clearOrDeleteAssociations(
```

Comment:
Out of curiosity, if a person doesn't have auth to clear or delete an association entity but has auth to delete the entity to which it is attached, will we end up with associations attached to nothing?

Reply:
That is exactly what we are tying to avoid :)

When we do a simple delete (not neighbors delete) of non-association entities
1. We first check authorizations on the requested entity set + properties (and fail call if checks fail)
2. Then we call this function, which also first checks authorizations on connected association entities and just then deletes association entities and the edges connected to them
3. As the last step we delete the normal entities

If we delete from an association entity set, we don't have to check connected associations, so we only delete its' edges in this function.

And in case of neighbors delete (when we delete from an entity set + its requested neighboring entity sets), we do basically the same: collect all neighbors, then check authorization, then delete associations and edges and as a last step delete entities.

Reply:
Gotcha, thank you for laying that out!